src/client.rs

```diff
@@ -215,7 +215,7 @@ impl<'a, T: Read + Write> AppendCmd<'a, T> {
     ///
     /// Note: be sure to set flags and optional date before you
     /// finish the command.
-    pub fn finish(&mut self) -> Result<()> {
+    pub fn finish(&mut self) -> Result<Appended> {
```

Comment:
Hmm, so, this is technically a breaking change. In practice it _should_ be very rare that someone relies on `()` specifically being the return type, but it is possible. Most likely if they have a call to `finish` somewhere without a trailing `;`, meaning its return value will end up being the evaluated value of the surrounding block as well.

How about we keep the return type of `finish` as-is, mark it as deprecated, and introduce a new method (bikeshed: `send` or `complete`) that returns `Result<Appended>`? `finish` can then just be a call to that method with a `.map(|_| ())` for the return value.